packages/test/stochastic-test-utils/src/markovChain.ts

```diff
@@ -0,0 +1,283 @@
+import { makeRandom } from "./random";
+import { IRandom } from "./types";
+
+abstract class MarkovChain {
+    public static readonly MARKOV_SENTENCE_BEGIN_KEY = "MARKOV_SENTENCE_BEGIN_KEY_01$#@%^#";
+    public static readonly MARKOV_SENTENCE_END_KEY = "MARKOV_SENTENCE_END_KEY_01$#@%^#";
+    public abstract initialize(predicitionPoints: string[][]): void;
+    public abstract generateSentence(...args: any): string;
+
+    protected static assumeWordLanguageSpacing(word: string): "SPACED" | "NO_SPACES" | "UNKNOWN" {
+        let spacedCount = 0;
+        let unspacedCount = 0;
+        for (let i = 0; i < word.length; i++) {
+            if (MarkovChain.getCharacaterSpacing(word.charAt(i)) === "SPACED") {
+                spacedCount++;
+            } else if (MarkovChain.getCharacaterSpacing(word.charAt(i)) === "NO_SPACES") {
+                unspacedCount++;
+            }
+        }
+
+        if (spacedCount >= unspacedCount) {
+            return "SPACED";
+        }
+        if (spacedCount < unspacedCount) {
+            return "NO_SPACES";
+        }
+        // Will return if spacedCount + unspacedCount === 0
+        return "UNKNOWN";
+    }
+
+    // Todo: Add support for more languages
+    protected static getCharacaterSpacing(character: string): "SPACED" | "NO_SPACES" | "UNKNOWN" {
+        // Latin without symbols & numbers
+        // range 1: ABCDEFGHIJKLMNO0050PQRSTUVWXYZ
```

Comment:
'0050' looks like an accidental paste?  It looks like it's propagated a couple of places.  Recommend grepping.

Reply:
Fixed.